tests/components/openweathermap/test_config_flow.py

```diff
@@ -0,0 +1,232 @@
+"""Define tests for the OpenWeatherMap config flow."""
+from asynctest import MagicMock, patch
+from pyowm.exceptions.api_call_error import APICallError
+from pyowm.exceptions.api_response_error import UnauthorizedError
+
+from homeassistant import data_entry_flow
+from homeassistant.components.openweathermap.const import (
+    CONF_LANGUAGE,
+    DEFAULT_FORECAST_MODE,
+    DEFAULT_LANGUAGE,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_MODE,
+    CONF_NAME,
+)
+
+from tests.common import MockConfigEntry
+
+CONFIG = {
+    CONF_NAME: "openweathermap",
+    CONF_API_KEY: "foo",
+    CONF_LATITUDE: 50,
+    CONF_LONGITUDE: 40,
+    CONF_MODE: DEFAULT_FORECAST_MODE,
+    CONF_LANGUAGE: DEFAULT_LANGUAGE,
+}
+
+VALID_YAML_CONFIG = {CONF_API_KEY: "foo"}
+
+
+async def test_form(hass):
+    """Test that the form is served with valid input."""
+    mocked_owm = _create_mocked_owm(True)
+
+    with patch(
+        "pyowm.weatherapi25.owm25.OWM25",
+        return_value=mocked_owm,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == SOURCE_USER
+        assert result["errors"] == {}
+
+        result = await hass.config_entries.flow.async_init(
```

Comment:
The flow can be continued with `hass.config_entries.flow.async_configure`.

Reply:
We normally patch the integration `async_setup_entry` function on entry create results to avoid setting up the whole integration, like is done in the test below.

Reply:
@freekode There are examples of patching `async_setup_entry` in `harmony`.